### PR TITLE
Created function to update property, created test for him

### PR DESCRIPTION
**Pull request by eskarlat**

### CODE REVIEW

File: contracts/Property/Property.sol [hunk 1]
```diff
@@ -51,9 +52,9 @@ contract Property is IProperty, OwnableUpgradeableImplementation, Pausable {
      * @dev modifier ensuring that the modified method is only called by the host of current property
      * @param propertyId - the identifier of the property
```
Comment (thcrnk) [inline contracts/Property/Property.sol]: Add hostAddress as @param

File: contracts/Property/Property.sol [hunk 2]
```diff
@@ -121,4 +135,37 @@ contract Property is IProperty, OwnableUpgradeableImplementation, Pausable {
         
 		return true;
 	}
+
+    function update(
+        bytes32 _propertyId,
+		bytes32 _marketplaceId,
+        address _hostAddress,
```
Comment (thcrnk) [inline contracts/Property/Property.sol]: Add newHost parameter, currently host can't be changed

File: contracts/Property/Property.sol [hunk 3]
```diff
@@ -93,14 +108,13 @@ contract Property is IProperty, OwnableUpgradeableImplementation, Pausable {
         uint _refundPercent,
         uint _daysBeforeStartForRefund,
         bool _isInstantBooking
-		) public onlyInactive(_propertyId) whenNotPaused returns(bool success)
+		) public onlyInactive(_propertyId) onlyMarketplace whenNotPaused returns(bool success)
 	{
         
         require(_marketplaceId != "");
```
Comment (thcrnk) [inline contracts/Property/Property.sol]: Move those requires in modifiers

File: contracts/Property/IProperty.sol [hunk 4]
```diff
@@ -18,6 +19,20 @@ contract IProperty is IOwnableUpgradeableImplementation, IPausable {
...
+        address _hostAddress,
+		uint _workingDayPrice,
+        uint _nonWorkingDayPrice,
+        uint _cleaningFee,
+        uint _refundPercent,
+        uint _daysBeforeStartForRefund,
+        bool _isInstantBooking
+    ) public returns(bool success);
+
+    function setMarketplace(address _marketplaceAddress) public;
```
Comment (thcrnk) [inline contracts/Property/IProperty.sol]: onlyOwner?

**Comment by eskarlat** [comment]
Момчета, как мога да се управя тази грешка: 

Error: The contract code couldn't be stored, please check your gas amount.
    at Object.callback (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:329221:46)
    at /usr/local/lib/node_modules/truffle/build/cli.bundled.js:39618:25
    at /usr/local/lib/node_modules/truffle/build/cli.bundled.js:331159:9
    at /usr/local/lib/node_modules/truffle/build/cli.bundled.js:175492:11
    at /usr/local/lib/node_modules/truffle/build/cli.bundled.js:314196:9
    at XMLHttpRequest.request.onreadystatechange (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:315621:13)
    at XMLHttpRequestEventTarget.dispatchEvent (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:70159:18)
    at XMLHttpRequest._setReadyState (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:70449:12)
    at XMLHttpRequest._onHttpResponseEnd (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:70604:12)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:70564:24)

File: contracts/Property/Property.sol [hunk 5]
```diff
@@ -7,7 +7,8 @@ import "./../Marketplace/IMarketplace.sol";
 
 contract Property is IProperty, OwnableUpgradeableImplementation, Pausable {
     
-    IMarketplace public MarketplaceContract; 
+    IMarketplace public MarketplaceContract;
```
Comment (thcrnk) [inline contracts/Property/Property.sol]: Change to private

File: contracts/Property/Property.sol [hunk 6]
```diff
@@ -7,7 +7,8 @@ import "./../Marketplace/IMarketplace.sol";
...
+    address public marketplaceAddress;
```
Comment (thcrnk) [inline contracts/Property/Property.sol]: Change to private

File: contracts/Property/Property.sol [hunk 7]
```diff
@@ -121,4 +171,40 @@ contract Property is IProperty, OwnableUpgradeableImplementation, Pausable {
...
+        bool _isInstantBooking,
+        address _newHost
+    ) public returns(bool success)
+    {
+        require(_hostAddress != address(0));
+        require(_newHost != address(0));
+
+        validateUpdate(_propertyId, _marketplaceId, _hostAddress);
+
+        MarketplaceContract = IMarketplace(msg.sender);
```
Comment (thcrnk) [inline contracts/Property/Property.sol]: This is already chacked in validateUpdate

File: contracts/Property/Property.sol [hunk 8]
```diff
@@ -121,4 +171,40 @@ contract Property is IProperty, OwnableUpgradeableImplementation, Pausable {
...
+        require(MarketplaceContract.isApprovedMarketplace(_marketplaceId));
```
Comment (thcrnk) [inline contracts/Property/Property.sol]: This is already chacked in validateUpdate

**Pull request merged**